bitwarden_license/src/app/providers/manage/events.component.ts

```diff
@@ -0,0 +1,183 @@
+
+import { ApiService } from 'jslib-common/abstractions/api.service';
+import { ExportService } from 'jslib-common/abstractions/export.service';
+import { I18nService } from 'jslib-common/abstractions/i18n.service';
+import { PlatformUtilsService } from 'jslib-common/abstractions/platformUtils.service';
+import { UserService } from 'jslib-common/abstractions/user.service';
+
+import { EventView } from 'jslib-common/models/view/eventView';
+
+import { EventService } from 'src/app/services/event.service';
+
+@Component({
+    selector: 'provider-events',
+    templateUrl: 'events.component.html',
+})
+export class EventsComponent implements OnInit {
+    loading = true;
+    loaded = false;
+    providerId: string;
+    events: EventView[];
+    start: string;
+    end: string;
+    dirtyDates: boolean = true;
+    continuationToken: string;
+    refreshPromise: Promise<any>;
+    exportPromise: Promise<any>;
+    morePromise: Promise<any>;
+
+    private providerUsersUserIdMap = new Map<string, any>();
+    private providerUsersIdMap = new Map<string, any>();
+
+    constructor(private apiService: ApiService, private route: ActivatedRoute, private eventService: EventService,
+        private i18nService: I18nService, private toasterService: ToasterService, private userService: UserService,
+        private exportService: ExportService, private platformUtilsService: PlatformUtilsService,
+        private router: Router) { }
+
+    async ngOnInit() {
+        this.route.parent.parent.params.subscribe(async params => {
+            this.providerId = params.providerId;
+            const provider = await this.userService.getProvider(this.providerId);
+            if (provider == null || !provider.useEvents) {
+                this.router.navigate(['/providers', this.providerId]);
+                return;
+            }
+            const defaultDates = this.eventService.getDefaultDateFilters();
+            this.start = defaultDates[0];
+            this.end = defaultDates[1];
+            await this.load();
+        });
+    }
+
+    async load() {
+        const response = await this.apiService.getProviderUsers(this.providerId);
+        response.data.forEach(u => {
+            const name = u.name == null || u.name.trim() === '' ? u.email : u.name;
+            this.providerUsersIdMap.set(u.id, { name: name, email: u.email });
+            this.providerUsersUserIdMap.set(u.userId, { name: name, email: u.email });
+        });
+        await this.loadEvents(true);
+        this.loaded = true;
+    }
+
+    async exportEvents() {
+        if (this.appApiPromiseUnfulfilled() || this.dirtyDates) {
+            return;
+        }
+
+        this.loading = true;
+
+        const dates = this.parseDates();
+        if (dates == null) {
+            return;
+        }
+
+        try {
+            this.exportPromise = this.export(dates[0], dates[1]);
+
+            await this.exportPromise;
+        } catch { }
```

Comment:
The events are handled by the `appApiAction`. But I've also added the `logService` which should make it easier to debug in the future.

I also took the opportunity to refactor it into a `BaseEventsComponent` since most of the code were duplicated between organizations and providers.